### PR TITLE
chore(setup): add SDK/project detection library

### PR DESCRIPTION
**Pull request by ffantl-ld**
**Describe the solution you've provided**

First layer of the guided `setup` command: the `internal/setup` detection library. Inspects a project directory and identifies the language/package manager and the most likely LaunchDarkly SDK (`Detector` interface, `FileDetector`, `KnownSDKs`).

Pure library with no consumer yet — the `setup` command wires it up later in this stack.

**Related issues**

Part of the `setup-ld` feature. Stacked PR — base is `setup-ld`.

**Requirements**

- [x] I have added test coverage for new or changed functionality

---

> [!NOTE]
> **Low Risk**
> New library-only code with no CLI integration yet; heuristics may mis-detect polyglot repos but failures are overridable via `--sdk-id`.
> 
> **Overview**
> Adds **`internal/setup`**, a filesystem-based project detector for the upcoming guided `ldcli setup` flow. **`FileDetector`** scans a directory and returns language, framework, package manager, recommended LaunchDarkly **`sdk_id`**, and an entry-point path, with **`EntryPointExists`** so callers only auto-edit files that actually exist.
> 
> Detection runs **Go → Python → Ruby → Java → Swift → .NET → Node** so backend manifests win over a root `package.json` used only for asset tooling. Node logic distinguishes Next.js (server SDK + `instrumentation.ts`), React Native, React, other browser frameworks, and generic Node; lockfiles drive npm/yarn/pnpm/bun. **`KnownSDKs`** lists manual override options when detection fails.
> 
> Heavy test coverage includes table-driven **project shapes** and edge cases (Next.js instrumentation vs pages, Android `MainActivity` search, ambiguous Go/Swift multi-binary layouts).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b5d737c707a6c197b2ea1c7781ceb7b742718d9a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: internal/setup/detector.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+	if result := detectJava(dir); result != nil {
+		return result, nil
+	}
+	if result := detectSwift(dir); result != nil {
+		return result, nil
+	}
+	if result := detectDotnet(dir); result != nil {
+		return result, nil
+	}
+	return nil, errors.New("could not detect project language from directory; try specifying --sdk-id manually")
```
Comment (ffantl-ld) [inline internal/setup/detector.go:66]: "could not detect project language from directory")

Should probably mean "Hey, here's a link to the docs and a blurb about how to do this install manually"

If we're doing a hybrid approach to setup, then we should definitely

**Comment by erangeles** [comment]
@ffantl-ld there is some valid bug bot issues flagged, wanted to make sure you were aware of them (either to fix now or punt on)

File: internal/setup/detector.go [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+	{ID: "android-client-sdk", Language: "Java", Name: "Android"},
+	{ID: "dotnet-server-sdk", Language: "C#", Name: ".NET"},
+	{ID: "swift-client-sdk", Language: "Swift", Name: "iOS/Swift"},
+	{ID: "ruby-server-sdk", Language: "Ruby", Name: "Ruby"},
+}
+
+// firstExistingIn returns the first candidate that exists as a file in dir,
+// or the last candidate if none exist (as a suggested path).
+// Returns an empty string if candidates is empty.
+func firstExistingIn(dir string, candidates []string) string {
```
Comment (erangeles) [inline internal/setup/detector.go]: suggestion: we should return whether the entry point actually exists so downstream steps can prompt the user instead


```go
type DetectResult struct {
      // ...
      EntryPoint       string `json:"entry_point"`
      EntryPointExists bool   `json:"entry_point_exists"`
}

// firstExistingIn returns the first candidate that exists in dir and true,
// or the last candidate and false as a suggestion if none exist.
func firstExistingIn(dir string, candidates []string) (string, bool) {
      if len(candidates) == 0 {
              return "", false
      }
      for _, c := range candidates {
              if _, err := os.Stat(filepath.Join(dir, c)); err == nil {
                      return c, true
              }
      }
      return candidates[len(candidates)-1], false
}
```

Then the wizard can branch: exists → inject; doesn't → ask the user for the entry file (or show the snippet).

**Comment by erangeles** [comment]
@ffantl-ld bump, just wanted to make sure you saw my earlier comments

File: internal/setup/detector.go (hunk 2 above)
Comment (ffantl-ld) [reply internal/setup/detector.go]: DetectResult now carries EntryPointExists. Good find. :)

**Review by erangeles — APPROVED** [review]

File: internal/setup/detector.go [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+					SDKID:            "android",
+					EntryPoint:       ep,
+					EntryPointExists: exists,
+				}
+			}
+			// Gradle layout: https://docs.gradle.org/current/userguide/building_java_projects.html
+			// Maven layout: https://maven.apache.org/guides/introduction/introduction-to-the-pom.html
+			ep, exists := entryPoint(dir, "src/main/java/Main.java",
+				findFileUnder(dir, "src/main/java", "Main.java", "Application.java", "App.java"),
+			)
```
[the remark this answers is not in the document]
Comment (ffantl-ld) [reply internal/setup/detector.go:334]: Will create a stack of fixes for growing the entry points of the detector.
